app/src/components/Modals/delete-city-modal.tsx

```diff
@@ -44,6 +49,27 @@ const DeleteCityModal: FC<DeleteCityModalProps> = ({
     setValue,
   } = useForm<{ password: string }>();
   const { t } = useTranslation(lng, "dashboard");
+  const [requestPasswordConfirm] = api.useRequestConfirmPasswordMutation();
```

Comment:
This is not the best approach to implement a password confirmation.
Usually you would use the same logic as a login, and require a recently created JWT token here (otherwise show a login form (which can be just a password prompt as you have here) or redirect the user).
Let's talk about this some more in a meeting.